lib/isEmail.js

```diff
@@ -68,6 +69,22 @@ function isEmail(str, options) {
       should be done in normalizeEmail
     */
     user = user.toLowerCase();
+
+    // Removing sub-address from username before gmail validation
+    var username = user.split('+')[0];
+
+    // Dots are not included in gmail length restriction
+    if (!(0, _isByteLength2.default)(username.replace('.', ''), { min: 6, max: 30 })) {
+      return false;
+    }
+
+    var _user_parts = username.split('.');
+    for (var i = 0; i < _user_parts.length; i++) {
+      if (!gmailUserPart.test(_user_parts[i])) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
Remove this line, or else everything after `+` isn't validated:

```
> validator.isEmail('foobar+ probably not <valid> @gmail.com')
true
```

Reply:
My bad.., forgot about gmail sub addresses..